### ISSUE TITLE
Installation fails due to conflicting Jinja2 version

### ISSUE THREAD

**Issue by NeolithEra** [InnoFang/what-digit-you-write#4]
Hi, users are unable to run **_what-digit-you-write_** due to dependency conflict with _**Jinja2**_ package. 
As shown in the following full dependency graph of **_what-digit-you-write_**, **_what-digit-you-write_** requires _**Jinja2==2.9.5**_，while **_flask 1.1.1_** requires _**Jinja2>=2.10.1**_.

According to pip’s _“first found wins”_ installation strategy, _**Jinja2==2.9.5**_ is the actually installed version. 
However, _**Jinja2==2.9.5**_ does not satisfy _**Jinja2>=2.10.1**_.

### Dependency tree
```
what-digit-you-write-master
| +-appdirs(version range:==1.4.3)
| +-click(version range:==6.7)
| +-flask(version range:>=0.12)
| | +-click(version range:>=5.1)
| | +-itsdangerous(version range:>=0.24)
| | +-jinja2(version range:>=2.10.1)
| | | +-markupsafe(version range:>=0.23)
| | +-werkzeug(version range:>=0.15)
| +-itsdangerous(version range:==0.24)
| +-jinja2(version range:==2.9.5)
| | +-markupsafe(version range:>=0.23)
| +-markupsafe(version range:==1.0)
| +-numpy(version range:==1.13.1)
| +-packaging(version range:==16.8)
| | +-pyparsing(version range:*)
| | +-six(version range:*)
| +-pillow(version range:==5.3.0)
| +-protobuf(version range:==3.2.0)
| +-pyparsing(version range:==2.2.0)
| +-six(version range:==1.10.0)
| +-tensorflow(version range:>=1.0.1)
| +-tensorflow-gpu(version range:>=1.0.1)
| +-werkzeug(version range:==0.12.1)
```

Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Solution

1.	Fix your direct dependency to be Jinja2>=2.9.5.
2.	Remove your direct dependency Jinja2, and use Jinja2 transitively introduced by **_flask_**.
3.	Fix your direct dependency to be flask==0.12. 


 I have checked the above revisions will not affect your downstream projects now.

Which solution do you prefer, 1 or 2 or 3?
@InnoFang Please let me know your choice. I can submit a PR to solve this issue.

**Comment by InnoFang** [comment]
> ### Solution
> 1. Fix your direct dependency to be Jinja2>=2.9.5.
> 2. Remove your direct dependency Jinja2, and use Jinja2 transitively introduced by **_flask_**.
> 3. Fix your direct dependency to be flask==0.12.
> 
> I have checked the above revisions will not affect your downstream projects now.
> 
> Which solution do you prefer, 1 or 2 or 3?
> @InnoFang Please let me know your choice. I can submit a PR to solve this issue.

Thank you for your proposal. I think Solution 2 will be better,

Looking forward to your PR.
